src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala

```diff
@@ -0,0 +1,185 @@
+package com.mesosphere.cosmos
+
+import java.io.OutputStream
+import java.io.PrintStream
+import java.net.URL
+import java.util.Properties
+import javassist.CannotCompileException
+import org.apache.curator.test.TestingCluster
+import sbt._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration._
+import scala.util.Random
+
+final class CosmosIntegrationTestServer(
+  javaHome: Option[String],
+  itResourceDirs: Seq[File],
+  oneJarPath: File
+) {
+  private val originalProperties: Properties = System.getProperties
+  private var process: Option[Process] = None           // scalastyle:ignore var.field
+  private var zkCluster: Option[TestingCluster] = None  // scalastyle:ignore var.field
+
+  def setup(logger: Logger): Unit = {
+    initZkCluster(logger)
+    val cmd = createServerCommand()
+    runProcess(logger, cmd)
+  }
+
+  def runProcess(logger: Logger, cmd: Seq[String]): Unit = {
+    logger.info("Starting cosmos with command: " + cmd.mkString(" "))
+
+    val prefix = "<<cosmos-server>> "
+    val run = Process(cmd).run(new ProcessLogger() {
+      override def buffer[T](f: => T): T = logger.buffer(f)
+
+      override def error(s: => String): Unit = logger.info(prefix + s)
+
+      override def info(s: => String): Unit = logger.info(prefix + s)
+    })
+    val fExitValue = Future(run.exitValue())
+    process = Some(run)
+    try {
+      waitUntilTrue(60.seconds) {
+        if (fExitValue.isCompleted) {
+          throw new IllegalStateException("Cosmos Server has terminated.")
+        }
+        canConnectTo(logger, new URL("http://localhost:9990/admin/ping"))
+      }
+    } catch {
+      case t: Throwable =>
+        cleanup()
+        throw t
+    }
+  }
+
+  def createServerCommand(): Seq[String] = {
+    val zkUri = zkCluster.map { c =>
+      val connectString = c.getConnectString
+      val zNodeNameSize = 10
+      val baseZNode = Random.alphanumeric.take(zNodeNameSize).mkString
+      s"zk://$connectString/$baseZNode"  // scalastyle:ignore multiple.string.literals
```

Comment:
Scalastyle has problems with interpolated strings. It seems to detect them as string literals containing a single double-quote character. Fortunately the error can be silenced with a single comment, but the reason for the comment isn't obvious.